src/main/java/com/google/devtools/build/lib/bazel/repository/DecompressorValue.java

```diff
@@ -59,9 +62,14 @@ private static String prepareErrorMessage(String prefix, Set<String> availablePr
       }
 
       public static Optional<String> maybeMakePrefixSuggestion(PathFragment pathFragment) {
-        return pathFragment.isMultiSegment()
-            ? Optional.of(pathFragment.getSegment(0))
-            : Optional.absent();
+        if (!pathFragment.isMultiSegment()) {
+          return Optional.empty();
+        }
+        String rawFirstSegment = pathFragment.getSegment(0);
+        // Users can only specify prefixes from Starlark, where strings always use UTF-8, so we
```

Comment:
> where strings always use UTF-8

How so? Can't a string literal embed bytes from an arbitrary encoding, and get stored directly in the low bytes of a Java String?

I don't know the significance of this code though. I'm just responding to the claim in the comment.

Reply:
You are right, I adjusted the comment.